general_autoencoder.py

```diff
@@ -316,11 +327,31 @@ def _fit_from_processed_data(self,
                 self.Z = self.encode(self.X)
                 
             self.Xr = self.decode(self.Z)
-            self.combined_loss, self.binary_loss, self.continuous_loss, self.reg_loss = self.get_loss()
+            self.combined_loss, self.binary_loss, self.continuous_loss, self.reg_loss = self.get_loss(self.X, self.Xr)
             self.optimizer = self.optimization_method(learning_rate=self.learning_rate).minimize(self.combined_loss)
+
             if self.uses_longitudinal_data:
-                self.combined_longitudinal_loss, self.binary_longitudinal_loss, self.continuous_longitudinal_loss, self.reg_longitudinal_loss = self.get_longitudinal_loss()
-                self.longitudinal_optimizer = self.optimization_method(learning_rate=self.learning_rate).minimize(self.combined_longitudinal_loss)
+
+                # now project Z0 forward to get Z1. 
+                # This requires multiplying the age components by longitudinal_ages1 / ages
+                self.Z1 = self.get_Z1_from_Z0(self.Z)
+
+                # reconstruct X1 from Z1.
+                self.Xr1 = self.decode(self.Z1)
+
+                _, self.binary_loss1, self.continuous_loss1, _ = self.get_loss(self.longitudinal_X1, self.Xr1)
+
+                self.combined_longitudinal_loss, self.binary_longitudinal_loss, \
```

Comment:
given that what's being done by get_longitudinal_loss is very simple (we just add them together and reweight) i wonder if it would be clearer to pull remove self.get_longitudinal_loss or pull it out of the subclass. maybe not.